table/update_schema.go

```diff
@@ -0,0 +1,532 @@
+package table
+
+import (
+	"github.com/apache/iceberg-go"
+)
+
+type UpdateSchema struct {
+	base                     *Metadata
+	schema                   *iceberg.Schema
+	idToParent               map[int]int
+	deletes                  []int
+	updates                  map[int]*iceberg.NestedField
```

Comment:
Do we need these to be pointers? Couldn't we just use `map[int]iceberg.NestedField`?

We are also missing tracking moving a column and adding a new column (additions should be tracked separately from updates)